tests/TestInitialConditions.py

```diff
@@ -342,4 +348,11 @@ def test_set_initial_geodetic_latitude(self):
         self.assertEqual(ref['Time'][0], 0.0)
         self.assertAlmostEqual(ref['Latitude Geodetic (deg)'][0], glat)
 
+    def PhiOrPsiAndGimbalLock(self, var, vars):
+        # Look out for quaternion singularity/gimbal-lock when ic/theta == 90
+        # and avoid checking psi and phi in this case
+        if (var['tag'] == 'psi' or var['tag'] == 'phi') and [x for x in vars if x['tag'] == 'theta' and x['value'] == 90]:
```

Comment:
Regarding the test `x['value'] == 90` : checking the equality of 2 floating point values is evil :-)
Please do something as
```python
abs(x['value']-90) < 1E-7
```

Reply:
@bcoconni less evil now with the new commit ;-)